src/buffer/out/sgrStack.hpp

```diff
@@ -0,0 +1,238 @@
+
+            return iOption;
+        }
+
+        TextAttribute _CombineWithCurrentAttributes(const TextAttribute& currentAttributes,
+                                                    const TextAttribute& savedAttribute,
+                                                    uint32_t validParts) noexcept // of savedAttribute
+        {
+            TextAttribute result = currentAttributes;
+
+            // From xterm documentation:
+            //
+            //  CSI # {
+            //  CSI Ps ; Ps # {
+            //            Push video attributes onto stack (XTPUSHSGR), xterm.  The
+            //            optional parameters correspond to the SGR encoding for video
+            //            attributes, except for colors (which do not have a unique SGR
+            //            code):
+            //              Ps = 1  -> Bold.
+            //              Ps = 2  -> Faint.
+            //              Ps = 3  -> Italicized.
+            //              Ps = 4  -> Underlined.
+            //              Ps = 5  -> Blink.
+            //              Ps = 7  -> Inverse.
+            //              Ps = 8  -> Invisible.
+            //              Ps = 9  -> Crossed-out characters.
+            //              Ps = 1 0  -> Foreground color.
+            //              Ps = 1 1  -> Background color.
+            //              Ps = 2 1  -> Doubly-underlined.
+            //
+            //  (some closing braces for people with editors that get thrown off without them: }})
+            //
+            // Attributes that are not currently supported are simply ignored.
+
+            if (_GraphicsOptionToFlag(DispatchTypes::GraphicsOptions::BoldBright) & validParts)
+            {
+                if (savedAttribute.IsBold())
+                {
+                    result.Embolden();
+                }
+                else
+                {
+                    result.Debolden();
+                }
+            }
+
+            if (_GraphicsOptionToFlag(DispatchTypes::GraphicsOptions::Underline) & validParts)
+            {
+                if (savedAttribute.IsUnderline())
+                {
+                    result.EnableUnderline();
+                }
+                else
+                {
+                    result.DisableUnderline();
+                }
+            }
+
+            if (_GraphicsOptionToFlag(DispatchTypes::GraphicsOptions::Negative) & validParts)
+            {
+                if (savedAttribute.IsReverseVideo())
+                {
+                    if (!result.IsReverseVideo())
+                    {
+                        result.Invert();
+                    }
+                }
+                else
+                {
+                    if (result.IsReverseVideo())
+                    {
+                        result.Invert();
+                    }
+                }
+            }
+
+            if (_GraphicsOptionToFlag(DispatchTypes::GraphicsOptions::SaveForegroundColor) & validParts)
```

Comment:
This list of attributes may need to be extended to support strikethrough/italic/the other extended text attributes introduced in #2917.

Reply:
Done.